migrator/migrations/m_180_to_m_181_move_to_blobstore/migration.go

```diff
@@ -0,0 +1,126 @@
+package m180tom181
+
+import (
+	"context"
+	"database/sql"
+	"os"
+
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/migrations/m_180_to_m_181_move_to_blobstore/schema"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/gorm/largeobject"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/utils"
+	"gorm.io/gorm"
+)
+
+const (
+	scannerDefBlobName = "/offline/scanner/scanner-defs.zip"
+)
+
+var (
+	scannerDefPath = "/var/lib/stackrox/scannerdefinitions/scanner-defs.zip"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 180,
+		VersionAfter:   &storage.Version{SeqNum: 181},
+		Run: func(databases *types.Databases) error {
+			err := moveToBlobs(databases.GormDB)
+			if err != nil {
+				return errors.Wrap(err, "moving persistent files to blobs")
+			}
+			return nil
+		},
+	}
+	log = logging.LoggerForModule()
+)
+
+func moveToBlobs(db *gorm.DB) (err error) {
+	ctx := sac.WithAllAccess(context.Background())
+	db = db.WithContext(ctx).Table(schema.BlobsTableName)
+	pgutils.CreateTableFromModel(context.Background(), db, schema.CreateTableBlobsStmt)
+
+	tx := db.Begin(&sql.TxOptions{Isolation: sql.LevelRepeatableRead})
+	if err = moveScannerDefination(tx); err != nil {
+		result := tx.Rollback()
+		if result.Error != nil {
+			return result.Error
+		}
+		return err
+	}
+
+	return tx.Commit().Error
+}
+
+func moveScannerDefination(tx *gorm.DB) error {
+	stat, err := os.Stat(scannerDefPath)
+	if err != nil {
+		if os.IsNotExist(err) || stat.IsDir() {
+			return nil
+		}
+		return err
+	}
+	modTime, err := timestamp.TimestampProto(stat.ModTime())
+	if err != nil {
+		return errors.Wrapf(err, "invalid timestamp %v", stat.ModTime())
+	}
+	fd, err := os.Open(scannerDefPath)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	defer utils.IgnoreError(fd.Close)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open %s", scannerDefPath)
+	}
+
+	// Prepare blob
+	blob := &storage.Blob{
+		Name:         scannerDefBlobName,
+		Oid:          0,
+		Length:       stat.Size(),
+		LastUpdated:  timestamp.TimestampNow(),
+		ModifiedTime: modTime,
+	}
+	los := largeobject.LargeObjects{DB: tx}
+
+	// Find the blob if it exists
+	var targets []schema.Blobs
+	result := tx.Limit(1).Where(&schema.Blobs{Name: scannerDefBlobName}).Find(&targets)
+	if result.Error != nil {
+		return result.Error
+	}
+
+	if len(targets) == 0 {
```

Comment:
Do we need to check and distinguish in this case? Migrations only run once and we can be guaranteed that no object will exist in the database because all of this is wrapped in a txn. I think it's a fair assumption that this does not exist and that we will always need to add and create

Reply:
Ah good point, do you think it'd be generally worthwhile to write out the version after each sequence going forward? Not in this PR of course